CONTRIBUTING.md

```diff
@@ -33,18 +79,25 @@ conda install -c conda-forge pre-commit
 To check and in most cases fix the code format, stage all your changes (`git add`) and run `pre-commit run`. To perform
 the checks automatically before every `git commit`, you can install them with `pre-commit install`.
 
+### Adding a New DataPipe
+
+When adding a new DataPipe, there are few things that need to be done to ensure it is working and documented properly.
+
+1. Testing - please add unit tests to ensure that the DataPipe is functioning properly. Here are the
+   [test requirements](https://github.com/pytorch/data/issues/106) that we have.
+2. Documentation - ensure that the DataPipe has docstring, usage example, and that it is added to the right category of
+   the right RST file to be rendered.
+3. Import - import the DataPipe in the correct `__init__.py` file.
+4. Interface - if the DataPipe has a functional form, make sure that is generated properly by `gen_pyi.py` into the
+   relevant interface file.
```

Comment:
Can you add a comment here. PPl can verify the `pyi` file by running `python setup.py develop`